### PR TITLE
txn: seperate the prewrite and commit details information to make it clear

### PR DESCRIPTION
**Pull request by cfzjywxk**
### What problem does this PR solve?

Issue Number: ref #34106

Problem Summary:
Separate the prewrite and commit details in commit information to make it clear.
Depends on the kv client change https://github.com/tikv/client-go/pull/567.

### Check List

Tests 

- [ ] Unit test

Side effects

Documentation

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -245,3 +245,5 @@ replace github.com/pingcap/tidb/parser => ./parser
```
Comment (hawkingrei) [inline go.mod:245]: when to update ```go.mod```, Please update bazel config to sync gomod into bazel.

File: util/execdetails/execdetails.go [hunk 2]
```diff
@@ -263,8 +271,20 @@ func (d ExecDetails) ToZapFields() (fields []zap.Field) {
 		if commitBackoffTime > 0 {
 			fields = append(fields, zap.String("commit_backoff_time", fmt.Sprintf("%v", strconv.FormatFloat(time.Duration(commitBackoffTime).Seconds(), 'f', -1, 64)+"s")))
 		}
-		if len(commitDetails.Mu.BackoffTypes) > 0 {
-			fields = append(fields, zap.String("backoff_types", fmt.Sprintf("%v", commitDetails.Mu.BackoffTypes)))
+		if len(commitDetails.Mu.PrewriteBackoffTypes) > 0 {
+			fields = append(fields, zap.String("Prewrite_"+BackoffTypesStr, fmt.Sprintf("%v", commitDetails.Mu.PrewriteBackoffTypes)))
+		}
```
Comment (sticnarf) [inline util/execdetails/execdetails.go:276]: Do we also need to add `CommitBackoffTypes` to `String` and `ToZapFields`?

Comment (cfzjywxk) [reply util/execdetails/execdetails.go:279]: Thanks for the reminding, they're added.

**Review by sticnarf — APPROVED** [review]

File: util/execdetails/execdetails.go [hunk 3]
```diff
@@ -86,8 +86,10 @@ const (
 	CommitBackoffTimeStr = "Commit_backoff_time"
 	// BackoffTypesStr means the backoff type.
 	BackoffTypesStr = "Backoff_types"
-	// SlowestCommitRPCDetailStr means the details of the slowest RPC during the transaction commit process.
-	SlowestCommitRPCDetailStr = "Slowest_commit_rpc_detail"
+	// SlowestPrewriteRPCDetailStr means the details of the slowest RPC during the transaction 2pc prewrite process.
+	SlowestPrewriteRPCDetailStr = "Slowest_prewrite_rpc_detail"
+	// CommitPrimaryRPCDetailStr means the details of the slowest RPC during the transaction 2pc commit process.
```
Comment (zyguan) [inline util/execdetails/execdetails.go:91]: Only commit primary?

File: util/execdetails/execdetails.go [hunk 4]
```diff
@@ -263,8 +274,23 @@ func (d ExecDetails) ToZapFields() (fields []zap.Field) {
...
+			fields = append(fields, zap.String(SlowestPrewriteRPCDetailStr, "total:"+strconv.FormatFloat(commitDetails.Mu.SlowestPrewrite.ReqTotalTime.Seconds(), 'f', 3, 64)+
+				"s, region_id: "+strconv.FormatUint(commitDetails.Mu.SlowestPrewrite.Region, 10)+
+				", store: "+commitDetails.Mu.SlowestPrewrite.StoreAddr+
+				", "+commitDetails.Mu.SlowestPrewrite.ExecDetails.String()+"}"))
+		}
+		if commitDetails.Mu.CommitPrimary.ReqTotalTime > 0 {
+			fields = append(fields, zap.String(CommitPrimaryRPCDetailStr, "{total:"+strconv.FormatFloat(commitDetails.Mu.SlowestPrewrite.ReqTotalTime.Seconds(), 'f', 3, 64)+
+				"s, region_id: "+strconv.FormatUint(commitDetails.Mu.SlowestPrewrite.Region, 10)+
+				", store: "+commitDetails.Mu.SlowestPrewrite.StoreAddr+
+				", "+commitDetails.Mu.SlowestPrewrite.ExecDetails.String()+"}"))
```
Comment (zyguan) [inline util/execdetails/execdetails.go:293]: Shall we extract a function for coverting `ReqDetailInfo` to string? (maybe better to impl `String()` method for `ReqDetailInfo`)

File: util/execdetails/execdetails.go (hunk 3 above)
Comment (cfzjywxk) [reply util/execdetails/execdetails.go:91]: For the query latency calculation of 2pc committed transactions, the primary key commit is enough.

File: util/execdetails/execdetails.go (hunk 4 above)
Comment (cfzjywxk) [reply util/execdetails/execdetails.go:293]: Better to abstract a string function for `ReqDetailInfo`, it needs to update client-go first maybe we could do it in the later prs.

**Review by zyguan — APPROVED** [review]

File: util/execdetails/execdetails.go (hunk 3 above)
Comment (zyguan) [reply util/execdetails/execdetails.go:91]: I mean the comment message is slightly inaccurate.

**Pull request merged**